### PR TITLE
[Woo POS] Fix for the payment flow crash

### PR DESCRIPTION
**Pull request by kidinov**
Closes:#12974
Closes:#12975
### Description

The PR: 
* Fixes the crash during payment flow in the POS
* Fixes the tracking of the onboarding events during POS during payments flow. Before it was tracked only on the connection flow
  * `woocommerceandroid_pos_payments_onboarding_shown` 
  *  `woocommerceandroid_pos_payments_onboarding_dismissed`

### Steps to reproduce

* More -> POS
* Add an item
* Collect payment
* Notice a crash

@AnirudhBhat @samiuelson could you please review the PR as this is quite dangerous

### Testing information

Pay attention to POS flows:
* Payments flow
* Onboarding screen, including going back from onboarding 
* Notice that the mentioned events are tracked both on the connection and payments flows now

###Retest IPP too!

### The tests that have been performed

Check the video

### Images/gif

https://github.com/user-attachments/assets/13966c4b-a309-4544-84dc-bcdf3fe3b9b3

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/244">21.2 ❄️</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by samiuelson — APPROVED** [review]
LGTM! 

Tested card payment in IPP and POS.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/onboarding/CardReaderOnboardingViewModel.kt [hunk 1]
```diff
@@ -89,6 +89,14 @@ class CardReaderOnboardingViewModel @Inject constructor(
 
     private var lastShownOnboardingState: CardReaderOnboardingState? = null
 
+    val isPos: Boolean
+        get() {
+            val cardReaderFlowParam = arguments.cardReaderOnboardingParam.cardReaderFlowParam
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/onboarding/CardReaderOnboardingViewModel.kt:94]: 💡 We should cover this logic with unit tests

**Review by AnirudhBhat — APPROVED** [review]
LGTM, I would wait for Samiuel's review as well.

Update: Oops, didn't notice Samiuel has already reviewed it. I would just suggest to add extensive unit tests to cover `isPOS` logic.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/onboarding/CardReaderOnboardingViewModel.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/payments/cardreader/onboarding/CardReaderOnboardingViewModel.kt:94]: Added tests here cafc0b8ef44712f6789774a4528e148076c12bb5

**Pull request merged**